### PR TITLE
Add server-only, client-only as devDependencies

### PR DESCRIPTION
**Pull request by Raghuboi**
Vercel recommends using the packages 'server-only' and 'client-only' to prevent unintended client usage of server code (and vice versa), we can use these packages to give other developers a build-time error if they ever accidentally import one of the server modules into a Client Component (or vice versa).

Read more @ [Keeping Server-Only Code out of Client Components (Poisoning)] (https://beta.nextjs.org/docs/rendering/server-and-client-components#keeping-server-only-code-out-of-client-components-poisoning)

### CODE REVIEW

**Comment by Raghuboi** [comment]
I realised that VSCode has made some formatting changes. Other than that, I have only added just the 1 import on top.

**Comment by Raghuboi** [comment]
> I realised that VSCode has made some formatting changes. Other than that, I have only added just the 1 import on top.

Maybe we can have a prettier config in the future

**Comment by xTrig** [comment]
> > I realised that VSCode has made some formatting changes. Other than that, I have only added just the 1 import on top.
> 
> Maybe we can have a prettier config in the future

Yeah ideally I'd like to have the same configuration as we had in student-net to keep things consistent.

According to https://beta.nextjs.org/docs/rendering/server-and-client-components, as long as our components are in the app/ directory, they are server components by default. We can't even use the 'use-client' directive if they are in that directory.

File: textbook-market/lib/mongodb.js [hunk 1]
```diff
@@ -1,34 +1,33 @@
 /*
     Example taken from Vercel's github, https://github.com/vercel/next.js/blob/canary/examples/with-mongodb/lib/mongodb.ts
 */
+import 'server-only'
```
Comment (xTrig) [inline textbook-market/lib/mongodb.js]: can we drop the changes to this file altogether?

**Review by xTrig — CHANGES_REQUESTED** [review]
Also we should probably add package-lock.json to gitignore as well, we don't need that on the repo.

**Comment by Raghuboi** [comment]
> > > I realised that VSCode has made some formatting changes. Other than that, I have only added just the 1 import on top.
> > 
> > 
> > Maybe we can have a prettier config in the future
> 
> Yeah ideally I'd like to have the same configuration as we had in student-net to keep things consistent.
> 
> According to https://beta.nextjs.org/docs/rendering/server-and-client-components, as long as our components are in the app/ directory, they are server components by default. We can't even use the 'use-client' directive if they are in that directory.

Are you sure? I thought 'use client' was meant to be used in the app/ directory, for whenever you want to use states (useState, useEffect, useContext). I've only ever used app/ in Next 13 and not both app/ and pages/ concurrently so might be something to do with that.

File: textbook-market/lib/mongodb.js (hunk 1 above)
Comment (Raghuboi) [reply textbook-market/lib/mongodb.js]: Sure

**Review by xTrig — APPROVED** [review]

**Pull request merged**